### ISSUE TITLE
"Building does not exist" option on "building type" quest

### ISSUE THREAD

**Issue by CTJoriginal** [Helium314/SCEE#499]
[SmartSelect_20231219_122200_SCEE.jpg](https://github.com/Helium314/SCEE/assets/48687492/5ef019ab-e3fd-4a96-8968-a0a90089e4ab)

**Use case**
Quite often, you run into buildings that are mapped but simply dont exist, ither they never did or they were demolished without a trace. In that case, you can only leave a note at this point.

**Proposed Solution**
Either expanding the "it's demolished" button to show a pop-up in this fashion:

---

*Are there any traces of demolished building left?*

*Please make sure that the building is not completely underground!*

YES - NO - CANCEL

---

Or, second option, just add another option to context menu on above image.

**Comment by mnalis** [comment]
> In that case, you can only leave a note at this point.

Well, you do seem to have expert mode enabled, so there is that `Show/edit tags` as an alternative to leaving note; but I get your point. 

> YES - NO - CANCEL

What should `NO` answer do, though? I see basically two possibilities:

- remove `building=*` tag and replace it with [razed:building=*](https://wiki.openstreetmap.org/wiki/Key:razed:*) (that should be very easy to implement)
- actually remove the building from the map (which is probably too complex for SCEE - that building might be closed way which has nodes shared with other ways [be it connected buildings or tunnels] , or it might be a part of multipolygon building relation etc).

I think (can't find the thread right now) the reasoning upstream was that **if the building is completely gone, then it is quite likely that other things around it might have gone / being repurposed too**, so roads / landuses / amenities / trees / street furniture etc. might need to be changed too, or new parking or playground or something else might have been put up there now, so leaving a Note (preferably with pictures) is actually the best thing to do, as it would likely be non-trivial editing session to update the situation (and much more suited for `JOSM` / `iD` / `Vespucci` than SC(EE)) 

Also, there is possibility that GPS is not so precise so user thinks they are looking at some place, when in fact they are not. In such cases too Note with explanation/pictures allows to clear/verify the situation on the aerial maps etc.

But of course, there are some situations where there was just one abandoned building that got torn down, and now there is really _nothing_ now where the building used to be. IMHO it boils down to _**how popular is that specific  (simplest) case**_? Does its popularity warrant special handling?
If it is popular enough so that such simple case (as opposed to more complex cases mentioned above) happens with some regularity that leaving Notes becomes burdensome, then it might indeed make sense to handle it via separate answer, sure. 

Although I can see a reasoning that still a Note might be preferable even for those simple cases  -- e.g. instead of just removing a building (or marking it as razed) one may want to mark the area as `landuse=brownfield` or `landcover=grass` or `highway=pedestrian`+`area=yes` or `natural=scrub` or whatever. It would be a shame to miss adding such information while one is on the ground and has best grasp of the situation.

---

**TL;DR:** Anyway, if it such option were to be introduced, I probably prefer separate `other answer` of `It's been razed - no visible traces remain` rather then followup question on `It's demolished`.
Although (as can probable be read between the lines above) I'm not seeing particular need for that in my little part of the world (e.g. I'd probably have such a clear use case which could be solved satisfactorily without a note maybe once per year at most)

**Comment by Helium314** [comment]
I agree with @mnalis. The frequent need for using a proper editor when dealing with removed buidlings is also why I only added the `demolished` answer, and not a `razed` answer.
Deleting anything else than nodes is currently not supported by SC(EE), and I don't plan to do anything here.

------

Also, please consider the text from the issue templates:
> Large images / videos can be very disturbing when reading, so please avoid attaching unnecessary screenshots and videos.
If you think they are needed, reduce the size or use thumbnails to keep the issue nicely readable.

**Comment by CTJoriginal** [comment]
I'm sorry for the late answer!

Yeah, after seeing your perspective, I agree that deleting buildings from the map could be potentially harmful.

It's probably best to leave it as-is or just add instructions for what to do in such case, for example, `If building was demolished with no visible traces, please leave a descriptive note, preferably with images of area.`

**Comment by Helium314** [comment]
How about using the normal _does not exist..._ other answer with choices of _demolished_, _other (leave note)_, and _cancel_?

**Comment by CTJoriginal** [comment]
That is actuall a great solution to all this!
